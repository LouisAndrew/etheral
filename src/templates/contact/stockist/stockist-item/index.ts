import { StockistItem, Props } from './stockist-item';

export { Props };

export default StockistItem;
