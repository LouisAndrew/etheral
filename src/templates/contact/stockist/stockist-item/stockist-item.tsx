import React from 'react';

import { Box, Text, Link } from 'rebass';

export type Props = {
    address: string;
    location: string;
    name: string;
    phoneNumber: string;
    web: string;
    zipCode: string;
};

const StockistItem: React.FC<Props> = ({
    name,
    address,
    phoneNumber,
    zipCode,
    web,
}) => {
    return (
        <Box
            as="li"
            sx={{
                listStyleType: 'none',
                textAlign: ['centerr', 'center', 'left'],
            }}
            my={[4]}
        >
            <Text
                fontSize={[1, 1, 2]}
                fontWeight="semiBold"
                fontFamily="body"
                mb={[3]}
            >
                {name}
            </Text>
            <Text
                as="p"
                fontSize={[0, 0, 1]}
                fontWeight="regular"
                fontFamily="body"
                width={['80%', '80%', '100%']}
                m="0 auto"
            >
                {address}
                <br />
                {zipCode}
                <br />
                {phoneNumber}
                <br />
            </Text>
            <Link
                href={`http://${web}`}
                mt={[1]}
                sx={{
                    fontSize: [0, 0, 1],
                    fontWeight: 'regular',
                    fontFamily: 'body',
                    textDecoration: 'none',
                }}
            >
                {web}
            </Link>
        </Box>
    );
};

export { StockistItem };
