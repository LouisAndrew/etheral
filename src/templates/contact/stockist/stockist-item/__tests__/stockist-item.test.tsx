import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { StockistItem } from '../stockist-item';

describe('StockistItem', () => {
    const stockistBdg = {
        address: '',
        location: 'Bandung',
        name: 'Toko C',
        phoneNumber: '1123',
        web: '',
        zipCode: '12',
    };

    const Element = <StockistItem {...stockistBdg} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
