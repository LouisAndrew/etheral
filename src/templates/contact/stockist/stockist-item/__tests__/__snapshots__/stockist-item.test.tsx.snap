// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockistItem matches snapshot 1`] = `
<li
  className="css-1t6fcji"
>
  <div
    className="css-m65dgr"
  >
    Toko C
  </div>
  <p
    className="css-uuqk9f"
  >
    
    <br />
    12
    <br />
    1123
    <br />
  </p>
  <a
    className="css-1ihmdtl"
    href="http://"
  >
    
  </a>
</li>
`;
